src/emailservice/email_server.py

```diff
@@ -193,7 +193,7 @@ def initStackdriverProfiling():
         project_id=os.environ.get('GCP_PROJECT_ID'),
         transport=AsyncTransport)
       tracer_interceptor = server_interceptor.OpenCensusServerInterceptor(sampler, exporter)
-  except (KeyError, DefaultCredentialsError):
+  except (KeyError, DefaultCredentialsError, OSError):
```

Comment:
- The `KeyError` is raised when the yaml doesn't define the environment variable `DISABLE_TRACING`. 
- I think we must log the `OSError` information to let the user know that `Trace has been enabled but the credentials cannot be found`. The above change might completely hide this detail

Reply:
@di Can you adapt your change for the above suggestion?

Reply:
Sorry, I'm not sure I follow. If an `OSError` is raised here, I don't believe it's possible to enable tracing, and this would be similar to the `DefaultCredentialsError`, no? Should we log an additional message for that exception as well?

Reply:
> * The `KeyError` is raised when the yaml doesn't define the environment variable `DISABLE_TRACING`.
> * I think we must log the `OSError` information to let the user know that `Trace has been enabled but the credentials cannot be found`. The above change might completely hide this detail

@Shabirmean I don't think this fix is designed to handle the case when `DISABLE_TRACING` isn't set (that's already caught by the service - when it's not set, tracing is disabled https://github.com/GoogleCloudPlatform/microservices-demo/blob/master/src/recommendationservice/recommendation_server.py#L107 ). Rather, I think @di was trying to fix a Cloud Ops client startup error that happens when `GOOGLE_APPLICATION_CREDENTIALS` are not set (If I'm understanding this right). 

@di your point about exception handling is well taking. will genericize the python startup exception handling to handle specific errors  as well as any uncaught exception.